texar/core/layers.py

```diff
@@ -664,6 +664,14 @@ def forward(self, input: torch.Tensor) -> torch.Tensor:  # type: ignore
                 layer_output = layer(input)
```

Comment:
(This comment is for line 660 but somehow I can't add a comment there.)

If `self._layers` is `None`, is it valid for `input` to be a tensor? To me it seems more reasonable if `input` is a list of tensors. This way the behaviors between cases w/ and w/o `self._layers` are consistent. Also, `torch.cat` requires the argument to be a list of tensors.

Reply:
`input` is the input to conv network or output of previous layers of the conv network, so as far as I know `input` is a `torch.Tensor`. The `if self._layers is None` handles a corner case where we allow `MergeLayer` to have no layers, in which case, we just return the input tensor. What we can do is, if the `if` condition holds true, we can set `layer_outputs = [inputs]`, in that way, `layer_outputs` is always a list.

Reply:
So when `self._layers` is `None`, `MergeLayer` acts as identity? I don't really think we need to handle this; we just don't construct the `MergeLayer` in that case, which saves an unnecessary function call during `forward`.

If we do need to support that case, then we should just return the `input` instead of executing the rest of the logic. And we should have a test case for that.

Reply:
Yes, it acts as an Identity operator. As discussed offline, we will provide this feature to maintain compatibility with `texar-TF`. I have added a test case to cover `self._layers = None` case.